cloud-minecraft/cloud-sponge7/src/main/java/cloud/commandframework/sponge7/CloudCommandCallable.java

```diff
@@ -0,0 +1,227 @@
+                    } else {
+                        source.sendMessage(MESSAGE_INTERNAL_ERROR);
+                        this.manager.getOwningPlugin().getLogger().error(
+                            "An unhandled exception was thrown during command execution",
+                            throwable
+                        );
+                    }
+                }
+            });
+        return CommandResult.success();
+    }
+
+    private Text formatMessage(final Throwable exc) {
+        if (exc instanceof TextMessageException) {
+            final Text response = ((TextMessageException) exc).getText();
+            if (response == null) {
+                return Text.of(TextColors.GRAY, "null");
+            } else if (response.getColor() == TextColors.NONE) {
+                return response.toBuilder().color(TextColors.GRAY).build();
+            } else {
+                return response;
+            }
+        } else {
+            return Text.of(TextColors.GRAY, exc.getMessage());
+        }
+    }
+
+    @Override
+    public List<String> getSuggestions(
+            final @NonNull CommandSource source,
+            final @NonNull String arguments,
+            final @Nullable Location<World> targetPosition
+    ) {
+        return this.manager.suggest(this.manager.getCommandSourceMapper().apply(source), this.formatCommand(arguments));
+    }
+
+    private String formatCommand(final String arguments) {
+        if (arguments.isEmpty()) {
+            return this.command.getName();
+        } else {
+            return this.command.getName() + " " + arguments;
+        }
+    }
+
+    @Override
+    public boolean testPermission(final @NonNull CommandSource source) {
+        return this.manager.hasPermission(this.manager.getCommandSourceMapper().apply(source), this.cloudCommand.getCommandPermission());
+    }
+
+    @Override
+    public @NonNull Optional<Text> getShortDescription(final @NonNull CommandSource source) {
+        final Optional<Text> richDesc = this.cloudCommand.getCommandMeta().get(SpongeMetaKeys.RICH_DESCRIPTION);
+        if (richDesc.isPresent()) {
+            return richDesc;
+        }
+
+        return this.cloudCommand.getCommandMeta().get(CommandMeta.DESCRIPTION).map(Text::of);
+    }
+
+    @Override
+    public @NonNull Optional<Text> getHelp(final @NonNull CommandSource source) {
+        final Optional<Text> richLongDesc = this.cloudCommand.getCommandMeta().get(SpongeMetaKeys.RICH_LONG_DESCRIPTION);
+        if (richLongDesc.isPresent()) {
+            return richLongDesc;
+        }
+
+        final Optional<Text> possible = this.cloudCommand.getCommandMeta().get(CommandMeta.LONG_DESCRIPTION).map(Text::of);
+        if (possible.isPresent()) {
+            return possible;
+        }
+
+        return this.getShortDescription(source);
+    }
+
+    @Override
+    public Text getUsage(final @NonNull CommandSource source) {
+        return Text.of(this.manager.getCommandSyntaxFormatter().apply(
```

Comment:
Is this correct? not sure

Reply:
https://github.com/Incendo/cloud/blob/e73c9a3bafe66a4c3846be57825ae4e8ef2ed868/cloud-core/src/main/java/cloud/commandframework/arguments/StandardCommandSyntaxFormatter.java#L53

I should probably clarify it in the JavaDoc. Basically, the list of arguments is meant to indicate what has already been supplied, i.e. the path that has taken you to the current command, whereas the nodes following that will explain what arguments can come *after* the current node.

Thus, I suppose the argument list should be empty (or the root), as nothing has been entered at this point.

Reply:
Ah, that makes sense. I've updated to use an empty list.